### PR TITLE
Set time_units

### PR DESCRIPTION
**Pull request by hyanwong**
No tests yet, draft to se if this is the right simple approach.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
LGTM. Does the ``time_units`` of the ancestors tree sequence get inherited by the final tree sequence?

**Comment by hyanwong** [comment]
> LGTM. 

Great

> Does the `time_units` of the ancestors tree sequence get inherited by the final tree sequence?

I will check in the tests. About to write now. Got held up by writing the workbook!

**Comment by hyanwong** [comment]
Tests done.

To avoid setting the time units on separate code paths, I changed `get_ancestors_tree_sequence` to return the tables (rather than converting the tables to a ts on the last line), then the conversion can happen in `store_output`. That seemed sensible anyway.

File: tests/test_inference.py [hunk 1]
```diff
@@ -1917,6 +1928,24 @@ def test_partial_bad_indexes(self):
             with pytest.raises(ValueError):
                 tsinfer.match_samples(sd, a_ts, indexes=bad_samples)
 
+    def test_time_units_default_uncalibrated(self):
```
Comment (jeromekelleher) [inline tests/test_inference.py:1931]: Need another test here to verify that passing time units to ``infer`` works also.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, just another test and update the changelog please.

File: tests/test_inference.py (hunk 1 above)
Comment (hyanwong) [reply tests/test_inference.py:1931]: OK, done. When this is merged I will close #679 and #469 but open up a new issue to discuss what I think is the correct approach.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -13,6 +13,12 @@
   bugs #965 and #967.
   (:issue:`643`, :pr:`657`, :user:`benjeffery`)
 
+**Breaking changes**:
```
Comment (jeromekelleher) [inline CHANGELOG.rst:16]: 0.2.3 is already released, need a new release 0.2.4. Date = 2022-06-xx

Comment (hyanwong) [reply CHANGELOG.rst]: Oops, sorry. Changed.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**